llama.h

```diff
@@ -195,15 +195,17 @@ extern "C" {
         LLAMA_KV_OVERRIDE_TYPE_INT,
         LLAMA_KV_OVERRIDE_TYPE_FLOAT,
         LLAMA_KV_OVERRIDE_TYPE_BOOL,
+        LLAMA_KV_OVERRIDE_TYPE_STR,
```

Comment:
If I remember correctly, there was some concern about overriding string KVs. That's why it wasn't implemented originally when @KerfuffleV2 added support for KV overrides. Need to search through the old discussions for a reference

Reply:
I think this is the reference: https://github.com/ggerganov/llama.cpp/pull/4092#issuecomment-1840910477

"*The only other potential issue with that is how it works with the "access metadata as string stuff", right now the two things don't interact at all so slaren's thing will use/print out the metadata without taking any overrides into account. I think printing it out like that is fine, overrides probably should affect accessing the actual data though.*"

Reply:
> I think this is the reference: https://github.com/ggerganov/llama.cpp/pull/4092#issuecomment-1840910477


Thanks, @jeximo. I read it twice. Honestly, I don't understand what's that means ^)). @slaren, could you please help

Reply:
I don't think that's relevant to this, @KerfuffleV2 was talking about the metadata values that are printed when loading the model and the ones returned with the `llama_model_meta_*` API functions. The overrides do not interact with this stuff, so these functions always print/return the original values.

Reply:
Thanks. @ggerganov, should we give it a try, what issue str overrides can cause at the today codebase ? Or I can remove string, but the value added of the PR becomes low.

Reply:
Yes, I had a wrong memory of a more complicated issue